### PR TITLE
#171205365 add cashier debit bank account

### PR DESCRIPTION
**Pull request by irfiacre**
#### What does this PR do?

 This PR adds new and modified files for the debit bank accounts endpoint.

#### Description of Task to be completed? 

- add cashier controller to controllers
- add debit account method on cashier controller
- add cashier validation and cashier tests

#### How should this be manually tested?
 - First, you clone the repo on your machine.
 - Open the folder in your text editor,  *preferably   `visual studio code`*
 - Now open your terminal, *for `visual studio code`    just click on    `terminal`* 
 - Now run `npm install`
 - Then  run `npm run dev`,
 - After now open `postman` and run endpoint using POST request *localhost:3000/transactions/`account-number`/debit* route; 

#### What are relevant pivotal tracker stories.
https://www.pivotaltracker.com/story/show/171205365
#### screenshots
image

account number not found
image

account  must be activated
image

Insufficient account balance
image

### CODE REVIEW

File: server/controllers/cashier.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    const oldBalance = rows2.balance;
+
+    if (oldBalance === 0 || oldBalance <= transaction.amount) {
+
+      return res.status(400).json({
+        status: 400,
+        error: 'INSUFFICIENT BALANCE ON THE ACCOUNT',
+      });
+    }
+    const newBalance = oldBalance - transaction.amount;
```
Comment (adafia) [inline server/controllers/cashier.js:49]: Good job, @irfiacre there's one more condition you need to handle. You need to make sure the transaction amount inputted is positive and not negative. Reach out for clarification if this isn't clear.

Comment (irfiacre) [reply server/controllers/cashier.js:49]: Should I convert the amount(negative) to a positive number or just return an error?

File: server/controllers/cashier.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    const getCashier = `SELECT * FROM users WHERE id= '${cashierId}';`;
+    const { rows: [cashierGet] } = await pool.query(getCashier);
+   
+
+
+    const transaction = {
+      date: req.body.date,
+      type: 'debit',
+      accountnNumber: req.params.accountNo,
```
Comment (adafia) [inline server/controllers/cashier.js]: Take a second look at this line. Did you mean to write "accountnNumber" or "accountNumber"?

File: server/controllers/cashier.js (hunk 1 above)
Comment (adafia) [reply server/controllers/cashier.js:49]: Return an error.

**Pull request merged**